File: java/src/main/java/vip/webmis/java/library/Down.java
package vip.webmis.java.library;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;

/* 下载类 */
public class Down {

  /* Blob方式 */
  public static byte[] fileBlob(String path, String filename){
    byte[] bytes = null;
    FileInputStream fs = null;
    try{
      File file = new File(path+filename);
      fs = new FileInputStream(file);
      bytes = new byte[(int)file.length()];
      fs.read(bytes);
      fs.close();
    }catch(IOException e){
      System.out.println("读取文件失败:"+e.getMessage());
    }finally{
      try { fs.close(); }catch(IOException e) { }
    }
    return bytes;
  }
  
}

File: phalcon/app/library/Down.php
<?php
namespace app\library;

/* 下载类 */
class Down{

  /* 导出Excel */
	static function excel($name='',$table='<table></table>'){
    // 输出数据
		echo $table;
		// 下载
		header("Content-type:application/octet-stream");
		header("Accept-Ranges:bytes");
    header("Content-type:application/vnd.ms-excel");
		header("Content-Disposition:attachment;filename=$name.xls");
		header("Pragma: no-cache");
		header("Expires: 0");
  }

  /* Blob方式 */
  static function fileBlob($path='',$filename=''){
    $file = $path.$filename;
    header('Content-type: application/octet-stream');
    header('Accept-Ranges: bytes');
    header('Accept-Length: '.filesize($file));
    header('Content-Disposition: attachment; filename='.$filename);
    return file_get_contents($file);
  }

}